log/log.c

```diff
@@ -337,24 +380,19 @@ void log_msgHandler(msg_t *msg, oid_t oid, unsigned long int rid)
 int log_write(const char *data, size_t len)
 {
 	size_t i = 0;
-	int overwrite = 0;
 	char c;
 
 	if (log_common.enabled != 0) {
 		proc_lockSet(&log_common.lock);
 
```

Comment:
Shoudn't `if (log_common.enabled != 0)` be checked once again after lock?

Reply:
I've analysed the code and I decided against the change. `log_disable` is only needed if we predict that emptying klog to the console by uart server or via dmesg command will not be possible anymore - pretty much kernel panic only. It is assumed that before printing the dying message `log_disable()` would be invoked and log safely printed to the terminal. We don't really care about other threads in this case and in case of hazard no dangerous condition arise - it's still safe to write to the log.

Reply:
do you think it worth to leave a comment in the code about it?

Reply:
I've added a brief comment